Lib/test/test_dataclasses/__init__.py

```diff
@@ -4869,5 +4869,125 @@ class A:
         self.assertEqual(fs[0].name, 'x')
 
 
+class TestZeroArgumentSuperWithSlots(unittest.TestCase):
+    def test_zero_argument_super(self):
+        @dataclass(slots=True)
+        class A:
+            def foo(self):
+                super()
+
+        A().foo()
+
+    def test_zero_argument_super_with_old_property(self):
+        @dataclass(slots=True)
+        class A:
+            def _get_foo(slf):
+                return slf.__class__
+
+            def _set_foo(slf, value):
+                self.assertIs(__class__, type(slf))
+
+            def _del_foo(slf):
+                self.assertIs(__class__, type(slf))
```

Comment:
I found the asymmetry between these methods confusing, and had to do some experimentation to clarify what exactly was being tested. I think there are three expressions that we use here: `__class__`, `slf.__class__`, and `type(self)`, and in all three methods of the property we ideally want to be sure that all three of those expressions have the same value.

I would find this clearer if all three methods simply had the same two lines in their body:
```
```
```suggestion
            def _get_foo(slf):
                self.assertIs(__class__, type(slf))
                self.assertIs(__class__, slf.__class__)

            def _set_foo(slf, value):
                self.assertIs(__class__, type(slf))
                self.assertIs(__class__, slf.__class__)

            def _del_foo(slf):
                self.assertIs(__class__, type(slf))
                self.assertIs(__class__, slf.__class__)
```